analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -359,6 +359,19 @@ void StringEmpty7(string path)
         {
             var s = string.IsNullOrEmpty(path) ? new string[] { } : path.Split('/');
         }
+
+        void BinaryOr(int i)
+        {
+            object o = null;
+            if (o==null | i == 0)
```

Comment:
Add a test case with the null check on the right hand side and short circuiting `if (i == 0 || o == null)`.

Reply:
|| is not a binary operation but a different CFG shape